subgraph/queries/tokens.py

```diff
@@ -38,6 +36,55 @@ def fetch_token_balances(
     """
     )
 
+
+def fetch_across_balances(block_number: int, chain: Network) -> Dict[str, int]:
+    if chain != Network.Ethereum:
+        return {}
+    increment = 1000
+    query = token_query()
+    continue_fetching = True
+    last_id = "0x0000000000000000000000000000000000000000"
+    multiplier = get_across_lp_multiplier()
+    console.log(f"Across lp multiplier {multiplier}")
+    across_balances = {}
+    client = make_gql_client("across")
+    try:
+        while continue_fetching:
+            variables = {
+                "firstAmount": increment,
+                "lastID": last_id,
+                "blockNumber": {"number": block_number},
+            }
+            next_page = client.execute(query, variable_values=variables)
+            if len(next_page["tokenBalances"]) == 0:
+                continue_fetching = False
+            else:
+                last_id = next_page["tokenBalances"][-1]["id"]
+                console.log(
+                    f"Fetching {len(next_page['tokenBalances'])} across balances"
+                )
+                for entry in next_page["tokenBalances"]:
+                    address = entry["id"].split("-")[1]
+                    amount = float(entry["balance"])
```

Comment:
Maybe should cast to `Decimal` instead?

Reply:
Techincally, this will be converted into Decimal when it is converted into a snapshot, so that isn't necessary here. At the moment most of the queries return floats or ints, but the snapshot class is what handles the conversion

Reply:
We can do an int conversion here instead of float